compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/CompiledPackageSymbolEnter.java

```diff
@@ -571,7 +571,8 @@ private void defineValueSpace(DataInputStream dataInStream, BFiniteType finiteTy
                 valueCPIndex = dataInStream.readInt();
                 UTF8CPEntry decimalEntry = (UTF8CPEntry) this.env.constantPool[valueCPIndex];
                 litExpr.value = decimalEntry.getValue();
-                litExpr.typeTag = TypeTags.DECIMAL;
+                // Type tag of the literal is float
+                litExpr.typeTag = TypeTags.FLOAT;
```

Comment:
Can you explain the logic scenario/behind here?

Reply:
We have two different kinds of tags. One is the literal's tag, which we refer by `literal.tag` and the other one is literal's type's tag referred by `literal.type.tag`. The former basically gives the tag of the actual literal, which is `float` in case of both decimal and float literals and the latter denotes the tag of the type of the literal, which should be `decimal` in case of a decimal literal. The same argument holds when a decimal value comes directly from the same module rather coming from a `balo`. 

However, earlier for the decimal constants those come from a `balo` the `literal.tag` was wrongly set. Hence, it is corrected with this PR.

Reply:
@pranavan15 - Shall we add this as a comment as well?

Reply:
@pranavan15 ,  can we refactor usage of `litExpr.typeTag` as we discussed offline?